server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java

```diff
@@ -145,36 +146,50 @@ public void enterCopyStatement(@NonNull CopyStatementContext ctx) {
 
   @Override
   public void exitCopyStatement(@NonNull CopyStatementContext ctx) {
+    collectAndAccumulateCopybookData(ctx.copySource(), true, retrieveCopybookStatementPosition(ctx), ctx.getSourceInterval());
+  }
+
+
+  @Override public void enterIncludeStatement(@NonNull IncludeStatementContext ctx) {
+    push();
+  }
+
+  @Override public void exitIncludeStatement(@NonNull IncludeStatementContext ctx) {
+    collectAndAccumulateCopybookData(ctx.copySource(), false, retrieveCopybookStatementPosition(ctx), ctx.getSourceInterval());
+  }
+
+  private void collectAndAccumulateCopybookData(CopySourceContext copySource, boolean applyReplacing, Locality copybookStatementPosition,  Interval sourceInterval) {
     if (!copybookProcessingMode.analyze) {
       pop();
       return;
     }
-    CopySourceContext copySource = ctx.copySource();
 
     String copybookName = retrieveCopybookName(copySource);
     Locality locality = retrievePosition(copySource);
     CopybookModel model = getCopyBookContent(copybookName, locality);
 
     String uri = model.getUri();
-    String rawContent = model.getContent();
+    String content = model.getContent();
 
-    String copybookId = randomUUID().toString();
-    String replacedContent = applyReplacing(rawContent, replacingClauses);
-    replacingClauses.clear();
+    if (applyReplacing) {
```

Comment:
Seems like this is an extra check. SQL INCLUDE doesn't allow replacing clauses, so nothing will be applied in any case. You may change it by checking `replacingClauses.isEmpty()` and don't pass the flag to this method.

Reply:
> Seems like this is an extra check. SQL INCLUDE doesn't allow replacing clauses, so nothing will be applied in any case. You may change it by checking `replacingClauses.isEmpty()` and don't pass the flag to this method.

done